src/analysis/vector/geometry_checker/qgsfeaturepool.h

```diff
@@ -24,39 +24,126 @@
 #include "qgis_analysis.h"
 #include "qgsfeature.h"
 #include "qgsspatialindex.h"
+#include "qgsvectorlayer.h"
 
 class QgsVectorLayer;
 
+/**
+ * \ingroup analysis
+ * A feature pool is based on a vector layer and caches features.
+ */
 class ANALYSIS_EXPORT QgsFeaturePool
 {
 
   public:
-    QgsFeaturePool( QgsVectorLayer *layer, double layerToMapUnits, const QgsCoordinateTransform &layerToMapTransform, bool selectedOnly = false );
+    QgsFeaturePool( QgsVectorLayer *layer, double layerToMapUnits, const QgsCoordinateTransform &layerToMapTransform );
+    virtual ~QgsFeaturePool() = default;
+
+    /**
+     * Retrieve the feature with the specified \a id into \a feature.
+     * It will be retrieved from the cache or from the underlying layer if unavailable.
+     * If the feature is neither available from the cache nor from the layer it will return false.
+     */
     bool get( QgsFeatureId id, QgsFeature &feature );
```

Comment:
For consistency with other classes, should this be renamed to "getFeature"? (and return QgsFeature/invalid QgsFeature instead of returning a bool)